src/components/plan-details-pages/PlanDetailsPage.tsx

```diff
@@ -0,0 +1,107 @@
+import { FormattedMessage } from '@edx/frontend-platform/i18n';
+import { zodResolver } from '@hookform/resolvers/zod';
+import {
+  Button,
+  Form,
+  Stack,
+  Stepper,
+} from '@openedx/paragon';
+import { Helmet } from 'react-helmet';
+import { useForm } from 'react-hook-form';
+import { useNavigate } from 'react-router-dom';
+
+import {
+  AuthenticatedUserField,
+  LicensesField,
+  NameAndEmailFields,
+} from '@/components/FormFields';
+import { PriceAlert } from '@/components/PriceAlert';
+import {
+  CheckoutStepKey,
+  CheckoutStepperPath,
+  PlanDetailsSchema,
+} from '@/constants/checkout';
+import useCheckoutFormStore from '@/hooks/useCheckoutFormStore';
+
+import '../Stepper/Steps/css/PriceAlert.css';
+
+const PlanDetailsPage: React.FC = () => {
+  const planFormData = useCheckoutFormStore((state) => state.formData.planDetails);
+  const formData = useCheckoutFormStore((state) => state.formData);
+  const { planDetailsRegistration, planDetailsLogin } = formData;
+  const isAuthenticated = planDetailsRegistration?.authenticated || planDetailsLogin?.authenticated;
+  const setFormData = useCheckoutFormStore((state) => state.setFormData);
+  // TODO: Once the user is logged in, use this field for authenticated user validation
+  // const { authenticatedUser } = useContext(AppContext);
+
+  const navigate = useNavigate();
+
+  const form = useForm<PlanDetailsData>({
+    mode: 'onTouched',
+    resolver: zodResolver(PlanDetailsSchema),
+    defaultValues: planFormData,
+  });
+  const {
+    handleSubmit,
+    formState: { isValid },
+  } = form;
+
+  const onSubmit = (data: PlanDetailsData) => {
+    // TODO: replace with existing user email logic
+    const randomExistingEmail = !!(Math.random() < 0.5 ? 0 : 1);
+    setFormData('planDetails', data);
```

Comment:
Should use shared `stepKey` variable:
```suggestion
    setFormData(stepKey, data);
```